### PR TITLE
✨ Add e2e style tests for Machine HealthChecking

### PR DESCRIPTION
**Pull request by JoelSpeed**
**What this PR does / why we need it**:

Adds more e2e style tests to the Machine Health Checking implementation.

Wanted to make sure these aren't flaky so ran them with `ginkgo -v -randomizeAllSpecs -untilItFails ./controllers/...` and it didn't fail on my machine in 10 attempts so I think they're good

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #1990

### CODE REVIEW

File: controllers/machinehealthcheck_controller_test.go [hunk 1]
```diff
@@ -206,6 +466,50 @@ func cleanupTestMachineHealthChecks(ctx context.Context, c client.Client) error
...
+		} else if err != nil {
+			return err
+		}
+		Eventually(func() error {
+			err := c.Get(ctx, util.ObjectKey(&m), &m)
+			if err != nil && apierrors.IsNotFound(err) {
+				return nil
+			} else if err != nil {
+				return err
+			}
```
Comment (neolit123) [inline controllers/machinehealthcheck_controller_test.go]: can be:
```
if err := c.Get(ctx, util.ObjectKey(&m), &m); err != nil {
	if apierrors.IsNotFound(err) {
		return nil
	}
	return err
}
```

File: controllers/machinehealthcheck_controller_test.go [hunk 2]
```diff
@@ -206,6 +466,50 @@ func cleanupTestMachineHealthChecks(ctx context.Context, c client.Client) error
...
+			m.SetFinalizers([]string{})
+			return c.Update(ctx, &m)
+		}, timeout).Should(Succeed())
+	}
+	return nil
+}
+
+func cleanupTestNodes(ctx context.Context, c client.Client) error {
+	nodeList := &corev1.NodeList{}
+	err := c.List(ctx, nodeList)
```
Comment (neolit123) [inline controllers/machinehealthcheck_controller_test.go]: scoping errors can be used to reduce the LOC in the diff:
```
if err := action(); err != nil {
...
}
```

File: controllers/machinehealthcheck_controller_test.go (hunk 1 above)
Comment (JoelSpeed) [reply controllers/machinehealthcheck_controller_test.go]: @neolit123 Thanks for the feedback, I did it slightly differently to your suggestion, WDYT?
```
if err := c.Get(ctx, util.ObjectKey(&m), &m); err != nil && apierrors.IsNotFound(err) {
    return nil
} else if err != nil {
    return err
}
```

Comment (neolit123) [reply controllers/machinehealthcheck_controller_test.go]: i prefer my method, but i can also now see the original was already used in number of places in this repo. no strong preferences on my side.

**Comment by enxebre** [comment]
Overall this looks great, thanks!

**Comment by vincepri** [comment]
/hold cancel

This looks good to go, thanks for the extra set of 👀 @cpanato !

**Pull request merged**